### PR TITLE
HPCC-16392 Avoid deregistering/re-registering memory callbacks.

### PR DESCRIPTION
**Pull request by jakesmith**
And a couple of other minor optimizations hit when running
child queries in a tight loop.

Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review

File: thorlcr/thorutil/thmem.cpp [hunk 1]
```diff
@@ -1904,17 +1902,9 @@ class CThorRowCollectorBase : public CSimpleInterface, implements roxiemem::IBuf
...
-        return spillPriority;
-    }
-    virtual unsigned getActivityId() const
-    {
-        return activity.queryActivityId();
-    }
     virtual bool freeBufferedRows(bool critical)
     {
-        if (!spillingEnabled())
+        if (!mmActivated || !spillingEnabled())
```
Comment (mckellyln) [inline thorlcr/thorutil/thmem.cpp]: Not sure I see why !mmActivated here.

Comment (jakesmith) [inline thorlcr/thorutil/thmem.cpp]: if not activated shouldn't do anything - so returns.

**Comment by jakesmith** [comment]
@mckellyln - I spotted an issue and have updated the PR - please re-review.

**Comment by mckellyln** [comment]
Looks fine.  I probably would have combined mmRegistered and mmActivated into a single variable as a mask.  @richardkchapman ok to merge.

**Pull request merged**